#!/usr/bin/env python
PACKAGE = "object_locator"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("min_disp", int_t, 0, "Minimum possible disparity value", 0, -1024, 1024)
gen.add("num_disp", int_t, 0, "Maximum disparity minus minimum disparity, must be divisable by 16", 224, 16, 1024)
gen.add("sad_size", int_t, 0, "Matched block size. It must be an odd number >=1 . Normally, it should be somewhere in the 3..11 range", 9, 1, 31)
gen.add("p1",       int_t, 0, "The first parameter controlling the disparity smoothness", 648,0,10000)
gen.add("p2",       int_t, 0, "TThe second parameter controlling the disparity smoothness. The larger the values are, the smoother the disparity is. P1 is the penalty on the disparity change by plus or minus 1 between neighbor pixels. P2 is the penalty on the disparity change by more than 1 between neighbor pixels. The algorithm requires P2 > P1", 2592,0,10000)
gen.add("disp12_max_diff", int_t, 0, "Maximum allowed difference (in integer pixel units) in the left-right disparity check. Set it to a non-positive value to disable the check", 1, -1, 256)
gen.add("pre_filter_cap",  int_t, 0, "Truncation value for the prefiltered image pixels", 31, 0, 63)
gen.add("uniqueness_ratio",int_t, 0, "Margin in percentage by which the best (minimum) computed cost function value should win the second best value to consider the found match correct. Normally, a value within the 5-15 range is good enough.", 10, 0, 100)
gen.add("speckle_window_size", int_t, 0, "Maximum size of smooth disparity regions to consider their noise speckles and invalidate. Set it to 0 to disable speckle filtering. Otherwise, set it somewhere in the 50-200 range", 900, 0, 1000)
gen.add("speckle_range",       int_t, 0, "Maximum disparity variation within each connected component. If you do speckle filtering, set the parameter to a positive value, it will be implicitly multiplied by 16. Normally, 1 or 2 is good enough.", 31, 0, 31)
gen.add("full_dp",             bool_t,0, "Set it to true to run the full-scale two-pass dynamic programming algorithm. It will consume O(W*H*numDisparities) bytes, which is large for 640x480 stereo and huge for HD-size pictures.", False)

exit(gen.generate(PACKAGE, "disparity_stage", "DisparityStage"))